internal/httpproxy/store.go

```diff
@@ -0,0 +1,177 @@
+package httpproxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// Mode definition
+type Mode string
+
+const (
+	// ModePassthrough mode
+	ModePassthrough Mode = "passthru"
+)
+
+// ErrorResponse definition
+type ErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// Request definition
+type Request struct {
+	Method  string      `json:"method"`
+	Path    string      `json:"path"`
+	Headers http.Header `json:"header"`
+	Query   url.Values  `json:"query"`
+	Body    []byte      `json:"body"`
+}
+
+// WriteResponse definition
+type WriteResponse struct {
+	StatusCode int               `json:"statuscode"`
+	Headers    http.Header       `json:"header"`
+	Body       WriteResponseBody `json:"body"`
+}
+
+// WriteResponseBody definition
+type WriteResponseBody struct {
+	Offset int `json:"offset"`
+}
+
+// Entry definition
+type Entry struct {
+	Offset   int           `json:"offset"`
+	Request  Request       `json:"request"`
```

Comment:
Call this `storeEntry` so its easier to read. Are you ever going to serialize this? Are the `json` annotations here really needed?